File: aspnet/Services/MandiriQrisConfig.cs
namespace MandiriQris.Services
{
    public class MandiriQrisConfig
    {
        public string Environment { get; set; } = "sandbox";
        public string ClientId { get; set; } = string.Empty;
        public string ClientSecret { get; set; } = string.Empty;
        public string MerchantNmid { get; set; } = string.Empty;
        public string MerchantName { get; set; } = "Toko Online";
        public string MerchantCity { get; set; } = "Jakarta";
        public int QrisExpiryMinutes { get; set; } = 5;
        public int Timeout { get; set; } = 30;

        public string BaseUrl => Environment == "production"
            ? "https://api.bankmandiri.co.id"
            : "https://sandbox.bankmandiri.co.id";
    }
}

File: java/src/main/java/com/mandiri/qris/config/MandiriQrisConfig.java
package com.mandiri.qris.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import lombok.Data;

@Data
@Configuration
@ConfigurationProperties(prefix = "mandiri.qris")
public class MandiriQrisConfig {
    
    private String environment = "sandbox";
    private String clientId;
    private String clientSecret;
    private String merchantNmid;
    private String merchantName = "Toko Online";
    private String merchantCity = "Jakarta";
    private Integer qrisExpiryMinutes = 5;
    private Integer timeout = 30;
    
    public String getBaseUrl() {
        return "production".equals(environment) 
            ? "https://api.bankmandiri.co.id" 
            : "https://sandbox.bankmandiri.co.id";
    }
}
